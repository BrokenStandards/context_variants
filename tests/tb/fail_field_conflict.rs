// Test case: field mentioned multiple times should be caught
use context_variants::variants;

#[variants(
    Create: requires(name).optional(name), // ERROR: name mentioned twice
    suffix = "Conflict"
)]
#[derive(Debug)]
struct ConflictTest {
    pub name: String,
    pub email: String,
}

fn main() {}
