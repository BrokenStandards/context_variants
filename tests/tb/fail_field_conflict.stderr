error: field 'name' mentioned multiple times: required, optional
 --> tests/ui/fail_field_conflict.rs:5:13
  |
5 |     Create: requires(name).optional(name), // ERROR: name mentioned twice
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: conflicting field specifications here

error: missing fields: email
 --> tests/ui/fail_field_conflict.rs:5:13
  |
5 |     Create: requires(name).optional(name), // ERROR: name mentioned twice
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: add .requires(email), .optional(email), .excludes(email), or .default(optional/required/exclude)
  = note: all fields must be specified here if `default(...)` is not set
